src/lib/parse.ts

```diff
@@ -0,0 +1,48 @@
+import { ParseError, UnsupportedFeatureError, WrongInputError } from "~/errors";
+import { AllowedBumps } from "~/types";
+
+function getName(title: string): string {
+  const match = /(bump|update) (?<name>(?:@[^\s]+\/)?[^\s]+) (requirement)?/i.exec(title);
+
+  if (match && match.groups && match.groups.name) {
+    return match.groups.name;
+  }
+
+  throw new ParseError("No valid dependancy 'name' found in PR title.");
+}
+
+function getRawVersion(title: string, target: "from" | "to"): string {
+  const regex =
+    /(?<version>(?<major>0|[1-9]\d*)(\.(?<minor>(0|[1-9]\d*))(\.(?<patch>(0|[1-9]\d*))(?:-((?<remainder>0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)?)?)/;
+
+  const matches = title.match(new RegExp(`${target} \\D*${regex.source}`));
+
+  if (matches && matches.groups && matches.groups.version && matches.groups.major) {
+    if (matches.groups.remainder) {
+      throw new UnsupportedFeatureError();
```

Comment:
`remainder` is a convenient way to throw if the bump is something like a prerelease of any voodoo stuff.